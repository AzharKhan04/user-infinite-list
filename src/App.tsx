import "./App.css";
import React from "react";
import { BrowserRouter, Routes, Route , Navigate } from "react-router-dom";
import AuthService from "./Services/auth.service";
import routes from "./routes";
import "bootstrap/dist/css/bootstrap.min.css";

const getAuth = () => {

  const authService:AuthService = new AuthService();
  return authService.isAuthenticated();
};

const ProtectedRoute: React.FunctionComponent<any> = ({ children }) => {

    const  authenticated  = getAuth();
    return authenticated === true
      ? children
      : <Navigate to="/login" replace />;
}

const App : React.FC<any> = () => {
  return (
    <BrowserRouter>
      <Routes>
        {
          routes.privateRoutes.map((route: any) => {
            return <Route key ={route.path} path={route.path} element={<ProtectedRoute>{route.component}</ProtectedRoute>
        } />;
          })
        }
        {routes.publicRoutes.map((route: any) => {
          return <Route key ={route.path} path={route.path} element={route.component} />;
        })}


      </Routes>
    </BrowserRouter>
  );
}

export default App;
