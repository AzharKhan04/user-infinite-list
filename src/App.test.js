import renderer from 'react-test-renderer';
import App from './App';

describe('App Component', () => {
  

it('App component renders correctly', () => {

  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();


});


})