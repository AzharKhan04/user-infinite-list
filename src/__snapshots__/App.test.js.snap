// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App component renders correctly 1`] = `null`;
